processing/src/main/java/org/apache/druid/query/scan/ScanQueryRunnerFactory.java

```diff
@@ -92,11 +93,15 @@ public QueryRunner<ScanResultValue> mergeRunners(
       ScanQuery query = (ScanQuery) queryPlus.getQuery();
       ScanQuery.verifyOrderByForNativeExecution(query);
 
-      // Note: this variable is effective only when queryContext has a timeout.
-      // See the comment of ResponseContext.Key.TIMEOUT_AT.
-      final long timeoutAt = System.currentTimeMillis() + QueryContexts.getTimeout(queryPlus.getQuery());
-      responseContext.putTimeoutTime(timeoutAt);
-
+      try {
+        // Note: this variable is effective only when queryContext has a timeout.
+        // See the comment of ResponseContext.Key.TIMEOUT_AT.
+        final long timeoutAt = System.currentTimeMillis() + QueryContexts.getTimeout(queryPlus.getQuery());
+        responseContext.putTimeoutTime(timeoutAt);
+      }
+      catch (NumberFormatException e) {
+        throw new BadQueryContextException(e);
```

Comment:
Maybe this exception should be thrown by the `QueryContexts.getTimeout()` method itself? I don't see why every caller should have to catch a `NumberFormatException`.